---
- id: ossindex:pypi:python-glanceclient:CVE-2013-4111
  title: "[CVE-2013-4111]  Improper Input Validation"
  description: The Python client library for Glance (python-glanceclient) before 0.10.0
    does not properly check the preverify_ok value, which prevents the server hostname
    from being verified with a domain name in the subject's Common Name (CN) or subjectAltName
    field of the X.509 certificate and allows man-in-the-middle attackers to spoof
    SSL servers via an arbitrary valid certificate.
  affected_package: python-glanceclient
  vulnerable_versions:
  - "<0.10.0"
  package_manager: pypi
  cve:
  - CVE-2013-4111
  disclosed_date: 2013-08-28
  created_date: 2013-08-28
  last_modified_date: 2017-01-15
  credit:
  - OSSIndex
  references:
  - http://lists.opensuse.org/opensuse-updates/2013-08/msg00019.html
  - http://rhn.redhat.com/errata/RHSA-2013-1200.html
  - http://secunia.com/advisories/54313
  - http://secunia.com/advisories/54525
  - http://www.ubuntu.com/usn/USN-2004-1
  - https://bugs.launchpad.net/ossa/%20bug/1192229
  - https://bugs.launchpad.net/ossa/+bug/1192229
  - https://github.com/openstack/python-glanceclient/blob/master/doc/source/index.rst
  source_url: https://ossindex.net/resource/package/8401237049/vulnerabilities
- id: snykio:pypi:python-glanceclient:CVE-2013-4111
  title: Man-in-the-Middle (MitM)
  description: |
    [`python-glanceclient`][1] is a OpenStack Image API Client Library.

    Affected versions of this package are vulnerable to Man-in-the-Middle
    (MitM) attacks. The Python client library for Glance
    (python-glanceclient) before 0.10.0 does not properly check the
    preverify\_ok value, which prevents the server hostname from being
    verified with a domain name in the subject\'s Common Name (CN) or
    subjectAltName field of the X.509 certificate and allows
    Man-in-the-Middle (MitM) attackers to spoof SSL servers via an arbitrary
    valid certificate.



    [1]: https://pypi.python.org/pypi/python-glanceclient
  affected_package: python-glanceclient
  vulnerable_versions:
  - "< 0.12.0"
  severity: medium
  package_manager: pypi
  cve:
  - CVE-2013-4111
  cwe:
  - CWE-20
  disclosed_date: 2013-06-18
  created_date: 2013-06-18
  last_modified_date: 2013-06-18
  credit:
  - Thomas Leaman
  references:
  - http://seclists.org/oss-sec/2013/q3/248
  - https://github.com/openstack/python-glanceclient/commit/822cd6
  source_url: https://snyk.io/vuln/SNYK-PYTHON-PYTHONGLANCECLIENT-40011
- id: victims:pypi:python-glanceclient:2013-4111
  title: 'python-glanceclient: Missing SSL certificate check'
  affected_package: python-glanceclient
  vulnerable_versions:
  - "<=0.11.0"
  patched_versions:
  - ">=0.12.0"
  severity: high
  package_manager: pypi
  cve:
  - CVE-2013-4111
  credit:
  - Victims CVE Database
  references:
  - http://seclists.org/oss-sec/2013/q3/248
  - https://bugs.launchpad.net/python-glanceclient/+bug/1192229
  - https://github.com/openstack/python-glanceclient/commit/822cd6
  source_url: https://github.com/victims/victims-cve-db/blob/master/database/python/2013/4111.yaml
